### PR TITLE
[cypress] user_logout

### PR DESCRIPTION
**Pull request by pbajjuri20**
Kiali User_logout

### CODE REVIEW

File: frontend/cypress/integration/common/kiali_logout.ts [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+        cy.get('button[type="submit"]').click()
+    }
+})
+
+And('user see console in URL', () => {
+    cy.url().should('include', 'console')
+})
+
+And('user clicks on admin', () => {
+    if (auth_strategy === 'openshift') {
```
Comment (ScriptingShrimp) [inline frontend/cypress/integration/common/kiali_logout.ts:54]: remove `if (auth_strategy === 'openshift')` condtion to pass on GH actions

File: frontend/cypress/integration/common/kiali_logout.ts [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+        cy.get('#user-dropdown-toggle').click() 
+    }
+})
+
+And('user logout successfully', () => {
+    cy.intercept('api/logout').as('logout')
+    if (auth_strategy === 'openshift') {
```
Comment (ScriptingShrimp) [inline frontend/cypress/integration/common/kiali_logout.ts:61]: remove `if (auth_strategy === 'openshift')` condtion to pass on GH actions

**Review by ScriptingShrimp — CHANGES_REQUESTED** [review]
there are still conditions remove `auth_strategy === 'openshift'` on places where they should not be

File: frontend/cypress/integration/common/kiali_logout.ts (hunk 1 above)
Comment (pbajjuri20) [reply frontend/cypress/integration/common/kiali_logout.ts]: removed the auth_strategy

**Comment by pbajjuri20** [comment]
I run this test case multiple times in the local, it run successfully. it is failing in the Git Hub Actions.

**Comment by pbajjuri20** [comment]
@ScriptingShrimp: I have executed in local with upstream cluster and it run perfectly.

It is failing in the "git-hub actions"

File: frontend/cypress/integration/common/kiali_logout.ts [hunk 3]
```diff
@@ -0,0 +1,19 @@
+import { Given, When, And, Then } from "cypress-cucumber-preprocessor/steps";
```
Comment (ScriptingShrimp) [inline frontend/cypress/integration/common/kiali_logout.ts:1]: `Given, When` imports are not used in this file, so there is no reason to import them

File: frontend/cypress/integration/featureFiles/kiali_logout.feature [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+
+  User wants to login to Kiali and see landing page
+  
+  @smoke
+  Scenario: Kiali logout successfully
+    Given user opens base url
+    And user clicks Log In With OpenShift
+    And user clicks my_htpasswd_provider
+    And user fill in username and password
+    And user see console in URL
```
Comment (ScriptingShrimp) [inline frontend/cypress/integration/featureFiles/kiali_logout.feature:11]: Sorry I did not noticed earlier: This is obsolete code, should be replaced with 
`Given user is at administrator perspective`

**Review by ScriptingShrimp — CHANGES_REQUESTED** [review]
please add newline in all files so you don't see red sign at the end of each file

File: frontend/cypress/integration/featureFiles/kiali_logout.feature (hunk 4 above)
Comment (pbajjuri20) [reply frontend/cypress/integration/featureFiles/kiali_logout.feature:11]: @ScriptingShrimp : It is taken last commit, I have update only ts file. I will update this file as well. Thanks for the review.

**Comment by pbajjuri20** [comment]
@ScriptingShrimp : I have updated the PR, according to your comments.

**Comment by lucasponce** [comment]
What's the status of this PR?

**Comment by pbajjuri20** [comment]
It is working perfectly in the local. It is not passing the Git-Hub Actions

Pavel requested some changes, I have changed them still it is not working. I will close this ticket ASAP.

**Comment by lucasponce** [comment]
This test is valid :-), I'd prefer if we could just fix it and use a selector that is updated with the changes:
image

Also, if it's platform-dependent, it's ok, as long as it has the proper conditional logic that can help to be run in any platform.

Thanks @pbajjuri20

**Comment by pbajjuri20** [comment]
@lucasponce: Actually, GitHub actions are using the "Anonymous" User. there is no LogOut button.

I'm writing the test case for User Logout. it is unable to find the "logout" button in the UI. 

that is the reason, it is failing :( 


user_1

**Comment by lucasponce** [comment]
> Actually, GitHub actions are using the "Anonymous" User. there is no LogOut button.

> I'm writing the test case for User Logout. it is unable to find the "logout" button in the UI.

It's not a problem; I mean that you could add some conditional logic to the test to execute them under some scenarios (i.e. non anonymous), or similar.

Then the suite would work under any pipeline.

**Review by ScriptingShrimp — APPROVED** [review]
LGTM!

**Pull request merged**